### DISCUSSION TITLE
Run backup script automatically everytime a new password is added

### DISCUSSION THREAD

**Discussion by ghost** [dani-garcia/vaultwarden#4273] [Q&A]
How would I go about implementing this?
I have already made a backup bash script that backups the vaultwarden folder properly using the correct method from [](https://github.com/dani-garcia/vaultwarden/wiki/Backing-up-your-vault).
If possible, I'd like to somehow run that automatically everytime a password is added to the vault.

**Comment by tessus** [comment]
Hmm, this seems rather excessive, but...

...I'd just use an update/insert trigger on the database (or better said a table within the db) which executes an external script (your backup script).
Depending on the database engine you use this might be possible.

**Reply by ghost** [reply]
After rethinking, I somewhat agree. My only fear was creating a new entry into the database that would somehow corrupt the database before my daily cronjob runs. But now that I think about it, most service providers I am storing passwords for have a forgot password option with email authentication, as well as recovery emails.
Daily backups are good enough I think.
